src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -0,0 +1,466 @@
+            ("blob" | "clob" | "lob", Language::Rust) => "Vec<u8>".to_string(),
+            ("blob" | "clob" | "lob", Language::Java) => "byte[]".to_string(),
+            ("decimal" | "timestamp" | "number", _) => {
+                unimplemented!("Decimal, Number and Timestamp aren't support yet!")
+            }
+            ("list" | "struct" | "sexp" | "document", _) => {
+                unimplemented!("Generic containers aren't supported yet!")
+            }
+            (_, _) => name.to_case(Case::UpperCamel),
+        }
+    }
+
+    /// Returns true if its a built in type otherwise returns false
+    pub fn is_built_in_type(name: &str) -> bool {
+        matches!(
+            name,
+            "int" | "i64" | "String" | "bool" | "boolean" | "byte[]" | "Vec<u8>" | "float" | "f64"
+        )
+    }
+
+    /// Represents a tera filter that converts given tera string value to upper camel case
+    /// Returns error if the given value is not a string
+    /// For more information: https://docs.rs/tera/1.19.0/tera/struct.Tera.html#method.register_filter
+    pub fn upper_camel(
+        value: &tera::Value,
+        _map: &HashMap<String, tera::Value>,
+    ) -> Result<tera::Value, tera::Error> {
+        Ok(tera::Value::String(
+            value
+                .as_str()
+                .ok_or(tera::Error::msg("Required string for this filter"))?
+                .to_case(Case::UpperCamel),
+        ))
+    }
+
+    /// Generates code for given Ion Schema
+    pub fn generate_code(
+        language: Language,
+        schema: IslSchema,
+        output: &Path,
+    ) -> CodeGenResult<()> {
+        // this will be used for Rust to create mod.rs which lists all the generates modules
+        let mut modules = vec![];
+        let mut module_context = tera::Context::new();
+        let mut code_gen_context = CodeGenContext::new(language);
+
+        // Register a tera filter that can be used to convert a string to upper camel case
+        code_gen_context
+            .tera
+            .register_filter("upper_camel", Self::upper_camel);
+
+        // Create a module for storing all the generated code in the output directory
+        fs::create_dir(output.join("ion_data_model"))?;
+
+        for isl_type in schema.types() {
+            // Initialize the data model as None for each type
+            // this will be filled with appropriate data model when code is generated for the type
+            code_gen_context.data_model = None;
+            Self::generate_data_model(output, &mut modules, &mut code_gen_context, isl_type)?;
+        }
+
+        if code_gen_context.language == Language::Rust {
+            module_context.insert("modules", &modules);
+            let rendered = code_gen_context
+                .tera
+                .render("rust/mod.templ", &module_context)?;
+            let mut file = File::create(output.join("ion_data_model/mod.rs"))?;
+            file.write_all(rendered.as_bytes())?;
+        }
```

Comment:
The fact that this is Rust-specific stands out. Eventually we should aim for the `generate_code` path to not contain anything specific to a target language; ideally `generate_code` would only call methods that discuss the target language in abstract, with methods on `Language` containing all of the language-specific bits.

Reply:
With the new change, context only stores data model property and is locally created for
each ISL type code generation.